packages/core/src/assert.ts

```diff
@@ -42,3 +49,56 @@ export function assertEvent<
     );
   }
 }
+
+function matchesEventDescriptor(
```

Comment:
don't we already have a good chunk of this in `getCandidates`? it would be great if this logic would be shared between both places